File: contests/gknot11/7.cpp
#include<cstdio>
#include<cmath>
int main()
{
	double p = (1.0+sqrt(5.0))/2.0;
	int i, c, x, y;

	i = c = 0;

	while(1)
	{
		x = floor(p*(double)i);
		y = floor(p*p*(double)i);

		if(x > 100000000)
			break;

		if( x == y)
			c+=1;
		else
			c+=2;

		i++;
	}

	printf("%d\n", c);
}

File: contests/gknot11/7.py
import math;
p = (1.0+math.sqrt(5.0))/2.0;

i = 0.0;
c = 0;
while True:
	x = math.floor(p*i);
	y = math.floor(p*p*i);

	if y > 100000000:
		break

	c+=1;

	i += 1.0;

print c;
